src/test/java/keycontacts/logic/commands/CancelLessonCommandTest.java

```diff
@@ -0,0 +1,119 @@
+import keycontacts.model.Model;
+import keycontacts.model.ModelManager;
+import keycontacts.model.UserPrefs;
+import keycontacts.model.lesson.CancelledLesson;
+import keycontacts.model.lesson.Date;
+import keycontacts.model.lesson.Time;
+import keycontacts.model.student.Student;
+
+public class CancelLessonCommandTest {
+    private Model model;
+
+    @BeforeEach
+    public void initialise() {
+        model = new ModelManager(getTypicalStudentDirectory(), new UserPrefs());
+    }
+
+    @Test
+    public void constructor_nullDate_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () ->
+                new CancelLessonCommand(INDEX_FIRST_STUDENT, null, new Time(VALID_START_TIME)));
+    }
+
+
+    @Test
+    public void constructor_nullTime_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () ->
+                new CancelLessonCommand(INDEX_FIRST_STUDENT, new Date(VALID_DATE), null));
+    }
+
+    @Test
+    public void constructor_nullIndex_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () ->
+                new CancelLessonCommand(null, new Date(VALID_DATE), new Time(VALID_START_TIME)));
+    }
+
+    @Test
+    public void execute_indexOutOfBounds_failure() {
+        Index outOfBoundsIndex = Index.fromOneBased(model.getFilteredStudentList().size() + 1);
+        CancelLessonCommand command = new CancelLessonCommand(outOfBoundsIndex, new Date(VALID_DATE),
+                new Time(VALID_START_TIME));
+
+        assertThrows(CommandException.class,
+                Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX, () -> command.execute(model));
+    }
+
+    @Test
+    public void execute_lessonNotFound_throwsCommandException() {
+        CancelLessonCommand command = new CancelLessonCommand(INDEX_FIRST_STUDENT, new Date(VALID_DATE),
+                new Time(VALID_START_TIME));
+
+        assertThrows(CommandException.class,
+                String.format(CancelLessonCommand.MESSAGE_LESSON_NOT_FOUND, Messages.format(ALICE)), () ->
+                        command.execute(model));
+    }
+
+    @Test
+    public void execute_validInputs_success() throws CommandException {
+        Date aliceDate = new Date("14-10-2024");
+        Time aliceTime = new Time("12:00");
+        CancelLessonCommand command = new CancelLessonCommand(INDEX_FIRST_STUDENT, aliceDate, aliceTime);
+        Student studentToUpdate = model.getFilteredStudentList().get(INDEX_FIRST_STUDENT.getZeroBased());
+        CancelledLesson cancelledLesson = new CancelledLesson(aliceDate);
+        Student expectedUpdatedStudent = studentToUpdate.withAddedCancelledLesson(cancelledLesson);
+        CommandResult result = command.execute(model);
+        String expectedMessage = String.format(CancelLessonCommand.MESSAGE_SUCCESS,
+                aliceDate, aliceTime, Messages.format(expectedUpdatedStudent));
+        assertEquals(expectedMessage, result.getFeedbackToUser());
+    }
```

Comment:
You are not checking for changes to the model. Please see `void execute_validIndexAndPianoPiece_success()` under `AssignPiecesCommandTest.java` and follow. If you do not want to move to using assertCommandSuccess, you will follow the same steps and add `assertEquals(expectedModel, model);` at the end.